dialog/file.go

```diff
@@ -35,13 +35,14 @@ type fileDialog struct {
 
 // FileDialog is a dialog containing a file picker for use in opening or saving files.
 type FileDialog struct {
-	save             bool
-	callback         interface{}
-	onClosedCallback func(bool)
-	filter           storage.FileFilter
-	parent           fyne.Window
-	dialog           *fileDialog
-	dismissText      string
+	save              bool
+	callback          interface{}
+	onClosedCallback  func(bool)
+	filter            storage.FileFilter
+	parent            fyne.Window
+	dialog            *fileDialog
+	dismissText       string
+	StartingDirectory string
```

Comment:
I think we need a URI based approach really, paths may not work across platforms